src/js/mixins/QueryParamsMixin.jsx

```diff
@@ -79,6 +79,10 @@ var QueryParamsMixin = {
     }
 
     router.transitionTo(router.getCurrentPathname(), {}, queryParams);
+  },
+
+  decodeArray: function (array) {
```

Comment:
Exactly, it is meant to decode ALL "array-like" values, because we encode every array-like value this way automatically. So it should lead to that assumption, this is by intention. 

`decodeLabelsQueryParam` is misleading, Our function only decodes a param-_value_ given by the functions argument, not the param itself, this was done at another place already before. And this argument has to be an array-like value.